nntrainer/tensor/cache_pool.cpp

```diff
@@ -106,23 +136,41 @@ void CachePool::invalidate(unsigned int id) {
   }
 }
 
+unsigned int CachePool::requestMemory(size_t bytes, unsigned int start_time,
+                                      unsigned int end_time,
+                                      std::vector<unsigned int> exec_order,
+                                      TensorLifespan lifespan) {
+  auto id = MemoryPool::requestMemory(bytes, start_time, end_time, exec_order,
+                                      lifespan);
+
+  const CachePolicy policy = convertTensorLifespanToCachePolicy(lifespan);
+
+  policies.push_back(policy);
+
+  NNTR_THROW_IF(id != policies.size(), std::runtime_error)
+    << "Invalid requqestMemory call exist";
+
+  return id;
+}
+
 std::shared_ptr<MemoryData<float>> CachePool::getMemory(unsigned int id) {
   NNTR_THROW_IF(!swap_device->isOperating(), std::invalid_argument)
     << "Allocate memory before allocation";
 
-  int offset = getMemoryOffset().at(id - 1);
+  size_t offset = getMemoryOffset().at(id - 1);
   size_t len = getMemorySize().at(id - 1);
   auto exe_order = getMemoryExecOrder().at(id - 1);
+  auto policy = getCachePolicy().at(id - 1);
```

Comment:
how about add ```getCachePolicy(unsigned int id)```?

Reply:
OK. It's also for getMemorySize,and others. I'll modify all apis in later PR.